include/odp/api/spec/schedule.h

```diff
@@ -340,6 +340,34 @@ int odp_schedule_capability(odp_schedule_capability_t *capa);
  */
 odp_schedule_group_t odp_schedule_group_create(const char *name,
 					       const odp_thrmask_t *mask);
+/**
+ * Initialize schedule group parameters
+ *
+ * Initialize an odp_schedule_group_param_t to its default values.
+ *
+ * @param[out] param  Pointer to parameter structure
+ */
+void odp_schedule_group_param_init(odp_schedule_group_param_t *param);
+
+/**
+ * Schedule group create with parameters
+ *
+ * Otherwise like odp_schedule_group_create() but additionally accepts a set of
+ * parameters.
+ *
+ * @param name    Name of the schedule group or NULL. Maximum string length is
+ *                ODP_SCHED_GROUP_NAME_LEN, including the null character.
+ * @param mask    Thread mask
+ * @param param   Schedule group parameters
+ *
+ * @return Schedule group handle
+ * @retval ODP_SCHED_GROUP_INVALID on failure
+ *
+ * @see odp_schedule_group_create
+ */
+odp_schedule_group_t odp_schedule_group_create2(const char *name,
+						const odp_thrmask_t *mask,
+						const odp_schedule_group_param_t *param);
```

Comment:
Similarly to odp_cls_pmr_create_opt(), the function name could be odp_schedule_group_create_opt() (i.e. "create with options") or odp_schedule_group_create_prm() ("create with params").

Reply:
We discussed this internally and this was what we ended up with. There's merit in using "version numbering" naming, there are no unwanted connotations and easier to add yet additional versions of functions if needed.

We can discuss this further offline.